docs/sources/get-started/components.md

```diff
@@ -217,24 +253,87 @@ The query frontend splits larger queries into multiple smaller queries, executin
+The query frontend queuing mechanism is used to:
+
+Query schedulers are **stateless**. However, due to the in-memory queue, it's recommended to run a more than one replica to keep the benefit of high availability. Two replicas should suffice in most cases.
+
 
 ## Querier
 
-The **querier** service handles queries using the [LogQL]({{< relref "../query" >}}) query
-language, fetching logs both from the ingesters and from long-term storage.
+The **querier** service is responsible for executing [Log Query Language (LogQL)]({{< relref "../query" >}}) queries.
+The querier can handle HTTP requests from the client directly (in "single binary" mode, or as part of the read path in "simple scalable deployment")
+or pull subqueries from the query frontend or query scheduler (in "microservice" mode).
 
+It fetches log data from both the ingesters and from long-term storage.
 Queriers query all ingesters for in-memory data before falling back to
 running the same query against the backend store. Because of the replication
 factor, it is possible that the querier may receive duplicate data. To resolve
 this, the querier internally **deduplicates** data that has the same nanosecond
 timestamp, label set, and log message.
 
-At read path, [replication factor]({{< relref "#replication-factor" >}}) also plays a role here. For example with `replication-factor` of `3`, we require that two queries to be running. 
 
+## Index Gateway
+
+The **index gateway** service is responsible for handling and serving metadata queries.
+Metadata queries are queries that look up data from the index. The index gateway is only used by "shipper stores",
+such as [single store TSDB]({{< relref "../operations/storage/tsdb" >}}) or [single store BoltDB]({{< relref "../operations/storage/boltdb-shipper" >}}).
+
+The query frontend queries the index gateway for volume of queries so it can make a decision on how to shard the queries.
+The queriers query the index gateway for chunk references for a given query so they know which chunks to fetch and query.
+
+The index gateway can run in `simple` or `ring` mode. In `simple` mode, each index gateway instance serves all indexes from all tenants.
+In `ring` mode, index gateways use a consistent hash ring to distribute and shard the indexes per tenant amongst available instances.
+
+
+## Compactor
+
+The **compactor** service is used by "shipper stores", such as [single store TSDB]({{< relref "../operations/storage/tsdb" >}})
+or [single store BoltDB]({{< relref "../operations/storage/boltdb-shipper" >}}), to compact the multiple index files produces by the ingesters
+and shipped to object storage into single index files per day and tenant. This makes index lookups more efficient.
+
+To do so, the compactor downloads the files from object storage in a regular interval, merges the them into a single one,
+uploads the newly created index, and cleans up the old files.
```

Comment:
should we also add a note here to run compactor as a single instance? (no replicas)
there is a note in retention page, but I think it's better suited here

Reply:
Added